pixl_core/src/core/uploader/base.py

```diff
@@ -47,10 +48,26 @@ def __init__(self, project_slug: str, keyvault_alias: Optional[str]) -> None:
     def _set_config(self) -> None:
         """Set the configuration for the uploader."""
 
+    def upload_dicom_and_update_database(self, study_id: str) -> None:
+        """
+        Upload the DICOM data, updating the database with an export datetime.
+        Child classes implement how to upload a dicom image, this is a template method
+        that ensures that the database interaction is always implemented
+        :param study_id: Orthanc Study ID
+        :raise: if the image has already been exported
+        """
+        pseudo_anon_image_id, project_slug = self._get_tags_by_study(study_id)
+        self.check_already_exported(pseudo_anon_image_id)
```

Comment:
Unexpected program flow. The assumption here is that we only want to export an study once. Could be a case where we've populate the queue twice by accident, before the studies have been fully processed. It's also useful to know if (as we found) mrn and accession number don't uniquely identify a study in the VNA. 

If we wanted to retry then we'd remove the `exported_at` for the specific studies we'd want because then the pixl CLI would publish them and the rest of the pipeline would accept them.